src/garage/torch/algos/vpg.py

```diff
@@ -228,18 +291,17 @@ def _compute_kl_constraint(self, obs):
             torch.Tensor: Calculated mean KL divergence.
 
         """
-        flat_obs = tu.flatten_batch(obs)
         with torch.no_grad():
-            old_dist = self._old_policy.forward(flat_obs)
+            old_dist = self._old_policy.forward(obs)
 
-        new_dist = self.policy.forward(flat_obs)
+        new_dist = self.policy.forward(obs)
 
         kl_constraint = torch.distributions.kl.kl_divergence(
             old_dist, new_dist)
 
         return kl_constraint.mean()
 
-    def _compute_policy_entropy(self, obs):
+    def compute_policy_entropy(self, obs):
```

Comment:
why does this need to be public?

Reply:
Because MAML class use this function.

Reply:
....but it's not an interface, which is why MAML has to use `# pylint-disable=protected-access`

please leave it as it was before.

Reply:
Sure, but I am a little bit confused.
I believe private function is only for internal use.
(and protected function is for internal and child class)

Since, this function is used in outside of the class, I think the function should be public.

Reply:
that's generally correct, but there's no well-defined interface here for these functions to implement. changing them to public doesn't solve that problem, it just makes pylint complain less.

making them public invites members of the public to rely on these functions implementing an interface which doesn't actually exist, and which we will probably change often in the future, breaking their code. a public function is an interface, and an interface is a promise. we can't promise these functions won't change.

we maintain the code so we can make sure that MAML doesn't mis-use them. i agree it would be better if MAML only used public interfaces, but that requires a larger refactor which we're not ready for yet.

Reply:
Okay, got it